README.md

```diff
@@ -36,42 +35,92 @@ Features
+
+### Using different classification methods
+
+1. running PermonSVM on 2 MPI processes with default settings (relaxed-bias classification, _l1_ hinge loss, C = 1, B = 1)
+   
+ 	```	bash 
+ 	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin
+ 	```
+  
+2. running PermonSVM on 2 MPI processes with penalty parameter C = 100 
+	
+	```	bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin 
+	-svm_C 100
+	```
+   
+3. running PermonSVM on 2 MPI processes with C = 0.01 and _l2_ hinge loss
+
+	```	bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin 
+	-svm_loss_type L2 -svm_C 1e-2
+	```
+  
+4. running PermonSVM on 2 MPI processes solving standard classification problem (binary mod 1), missclassification error quantification by _l2_ hinge loss, and C = 0.01
+	
+	```	bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin
+	-svm_loss_type L2 -svm_C 1e-2 -svm_binary_mod 1
+	```
+	
+### Hyperparameter optimization
+
+1. running PermonSVM on 2 MPI processes with hyperparameter optimization with default settings (_l1_ hinge loss function, relaxed-bias classification, grid-search log2C = [-2:1:2], k-fold cross validation on 5 folds)
+
+	```	bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin 
+	-svm_hyperopt 1
+	```
+   
+2. running PermonSVM on 2 MPI processes with grid-search on C = {0.1, 1, 10, 100} combined with cross validation on 3 folds that reuses a previous solution (warm start)
+
+	```bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin 
+	-svm_hyperopt 1 -svm_gs_logC_base 10 -svm_gs_logC_stride 1,2,1 -svm_nfolds 3 -cross_svm_warm_start 1
+	```
+  
+3. running PermonSVM on 2 MPI processes with grid search on C = {0.1, 1, 10, 100} and stratified k-fold cross validation on 3 folds with warm start
+
+	```bash
+	./runsvmmpi 2 -f_training $DATA_DIR/heart_scale.bin -f_test $DATA_DIR/heart_scale.t.bin 
+	-svm_hyperopt 1 -svm_gs_logC_base 10 -svm_gs_logC_stride 1,2,1 -svm_nfolds 3 -cross_svm_warm_start 1 
+	-svm_cv_type stratified_kfold
+	```
+   
+### Using precomputed Gramian matrix
+
+PermonSVM uses an implicit representation of the Gramian matrix by default. Sometimes, it is reasonable to compute inner products related to the Gramian explicitly, typically, when a number of features is disproportionately larger than a number of samples. For such cases, PermonSVM provides functionality allowing to load precomputed Gramian matrix 
```

Comment:
```suggestion
PermonSVM uses an implicit representation of the Gramian matrix by default.
Sometimes, it is reasonable to compute inner products related to the Gramian explicitly, typically, when a number of features is disproportionately larger than a number of samples.
For such cases, PermonSVM provides functionality allowing to load precomputed Gramian matrix.
```
(Just newlines, I know it doesn't matter for the output, but the raw MD is more readable and it's better for versioning.)